src/core/Table/Table.tsx

```diff
@@ -267,11 +274,29 @@ export const Table = <
         maxWidth: 48,
         columnClassName: 'iui-slot',
         cellClassName: 'iui-slot',
-        Header: ({ getToggleAllRowsSelectedProps }: HeaderProps<T>) => (
-          <Checkbox {...getToggleAllRowsSelectedProps()} />
-        ),
+        Header: ({ getToggleAllRowsSelectedProps }: HeaderProps<T>) => {
+          const onChange = () => {
+            const isAllEnabledRowsSelected =
+              instance.isAllRowsSelected ||
+              instance.rows.every(
+                (row) =>
+                  instance.state.selectedRowIds[row.id] ||
+                  isRowDisabled(row.original),
+              );
+            instance.toggleAllRowsSelected(!isAllEnabledRowsSelected);
```

Comment:
Interesting logic. From what I understand, this is a binary toggle. Where does the `indeterminate` state come from?

Reply:
From `...getToggleAllRowsSelectedProps()`

Reply:
So we need to override if we want it to be `checked` when all non-disabled rows are selected.

```tsx
checked: instance.rows.every((row) => row.isSelected || isRowDisabled(row.original)),
```

Reply:
Changed.